File: ambeth/Minerva.Client/ambeth/test/AbstractHelloWorldTest.cs
using De.Osthus.Ambeth.Config;
using De.Osthus.Ambeth.Helloworld.Service;
using De.Osthus.Ambeth.Ioc.Annotation;
using De.Osthus.Ambeth.Testutil;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace De.Osthus.Ambeth.Test
{
    [TestProperties(Name = ServiceConfigurationConstants.ServicePrefix, Value = "/helloworld")]
    [TestModule(typeof(HelloWorldTestModule))]
    [TestClass]
    public abstract class AbstractHelloWorldTest : AbstractRemoteTest
    {
        [Autowired]
        public IHelloWorldService HelloWorldService { protected get; set; }
    }
}

File: jambeth/jambeth-server-helloworld/src/test/java/de/osthus/ambeth/helloworld/AbstractHelloWorldTest.java
package de.osthus.ambeth.helloworld;

import de.osthus.ambeth.config.ServiceConfigurationConstants;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.testutil.AbstractInformationBusWithPersistenceTest;
import de.osthus.ambeth.testutil.SQLStructure;
import de.osthus.ambeth.testutil.TestProperties;

@SQLStructure("/create_hello_world_tables.sql")
@TestProperties(name = ServiceConfigurationConstants.mappingFile, value = "de/osthus/ambeth/helloworld/helloworld_orm.xml")
public abstract class AbstractHelloWorldTest extends AbstractInformationBusWithPersistenceTest
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;
}
